lib/kubeclient.rb

```diff
@@ -352,6 +352,8 @@ def create_faraday_client
       Faraday.new(url, options) do |connection|
         if @auth_options[:username]
           connection.request(:basic_auth, @auth_options[:username], @auth_options[:password])
+        elsif @auth_options[:bearer_token_file]
+          connection.request(:authorization, 'Bearer', -> { File.read(@auth_options[:bearer_token_file]).chomp })
```

Comment:
Ooh, neat, this make Faraday execute the proc on every request:

https://www.rubydoc.info/gems/faraday/Faraday/Request/Authorization,
https://github.com/lostisland/faraday/blob/main/lib/faraday/request/authorization.rb
> params (Array<String, Proc, call>) — parameters to build the Authorization header. If the type is `:basic`, then these can be a login and password pair. Otherwise, a single value is expected that will be appended after the type. This value can be a proc or an object responding to `.call`, in which case it will be invoked on each request.